projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,58 +1,77 @@
 import { Dictionary } from '@hypertrace/common';
-import { SelectOption } from '../../select/select-option';
+import { FilterOperator } from '../../filtering/filter/filter-operators';
 import { TableFilter } from '../table-api';
 
-export interface SelectControl {
-  placeholder?: string;
-  default?: SelectOption<TableControlOption>;
-  options: SelectOption<TableControlOption>[];
+export const enum TableControlOptionType {
+  Filter = 'filter',
+  Property = 'property',
+  Unset = 'unset'
 }
 
-export interface SelectChange {
-  select: SelectControl;
-  value: TableControlOption;
-}
+export type TableControlOption = TableUnsetControlOption | TableFilterControlOption | TablePropertyControlOption;
 
-export interface CheckboxControl {
+export interface TableFilterControlOption {
+  type: TableControlOptionType.Filter;
   label: string;
-  value: boolean;
-  options: TableCheckboxOptions;
+  metaValue: TableFilter;
 }
 
-export interface CheckboxChange {
-  checkbox: CheckboxControl;
-  option: TableControlOption<boolean>;
+export interface TableUnsetControlOption {
+  type: TableControlOptionType.Unset;
+  label: string;
+  metaValue: string;
 }
 
-export const enum TableControlOptionType {
-  Filter = 'filter',
-  Property = 'property',
-  UnsetFilter = 'unset-filter'
+export interface TablePropertyControlOption {
+  type: TableControlOptionType.Property;
+  label: string;
+  metaValue: Dictionary<unknown>;
 }
 
-export type TableControlOption<T = unknown> =
-  | TableUnsetFilterControlOption<T>
-  | TableFilterControlOption<T>
-  | TablePropertyControlOption<T>;
+/*
+ * Select Control
+ */
 
-interface TableControlOptionBase<T> {
-  value?: T;
+export interface SelectControl {
+  placeholder: string;
+  options: TableSelectControlOption[];
 }
-export interface TableUnsetFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.UnsetFilter;
-  metaValue: string;
+
+export interface SelectChange {
+  select: SelectControl;
+  values: TableSelectControlOption[];
 }
 
-export interface TableFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.Filter;
-  metaValue: TableFilter;
+export type TableSelectControlOption = TableFilterControlOption;
```

Comment:
do we really need so many different types? Like below where you specify metaValue type

```
export interface TableControlOption<T> {
  type: TableControlOptionType;
  label: string;
  metaValue: T;
}
}
```

Reply:
I didn't add them just for fun! 😂

But valid question. I explain a little more in [this](https://github.com/hypertrace/hypertrace-ui/pull/823/files#r626981848) comment.